qlty-check/src/executor.rs

```diff
@@ -603,26 +603,22 @@ impl Executor {
         cache_hits: &[IssuesCacheHit],
         invocations: &[InvocationResult],
         skip_errored_plugins: bool,
-    ) -> Vec<Issue> {
+    ) -> Result<Vec<Issue>> {
         let mut issues = vec![];
 
         for cache_hit in cache_hits {
             for issue in &cache_hit.issues {
                 issues.push(issue.to_owned());
 
                 if issues.len() >= MAX_ISSUES {
-                    warn!(
-                        "Maximum issue count of {} reached in cache, skipping further issues.",
-                        MAX_ISSUES
-                    );
-                    return issues;
+                    bail!("{}", Self::format_max_issues_error(&issues, " in cache"));
```

Comment:
We can simplify this -- we don't need to use a different message based on the issues being in the cache. Use the same message every time.